app/js/shared/loadGoogleLibraries.provider.js

```diff
@@ -0,0 +1,65 @@
+(function() {
+  'use strict';
+  angular.module('upcomingStops.shared').provider('LoadGoogleMapsApi', LoadGoogleMapsApiProvider);
+
+  function LoadGoogleMapsApiProvider() {
+      var myConfig = {
+        baseUrl: 'https://maps.googleapis.com/maps/api/js',
+        apiKey: null,
+        libraries: null,
+        language: 'en'
+      };
+
+      this.setConfig = function(config) {
+        angular.extend(myConfig, config);
+      }
+
+      var loadDeferred = null;
+      this.$get = function($window, $document, $q) {
+        if (loadDeferred !== null) {
+          return loadDeferred.promise;
+        }
+
+        loadDeferred = $q.defer();
+
+        function loadGMapsScript() {
+          if (myConfig.baseUrl == null) {
+            console.log('No Google Maps API URL defined, check Angular config!');
+            loadDeferred.reject('No Google Maps API URL');
+          }
+          else if (myConfig.apiKey == null) {
+            console.log('No Google Maps API Key defined, check Angular config!');
+            loadDeferred.reject('No Google Maps API Key');
+          }
+          else {
+            var fullUrl = myConfig.baseUrl + '?key=' + myConfig.apiKey;
+            if (myConfig.libraries != null) {
+              fullUrl += '&libraries=' + myConfig.libraries.join(',');
+            }
+            if (myConfig.language != null) {
+              fullUrl += '&language=' + myConfig.language;
+            }
+
+            fullUrl += '&callback=googlemapsapisloaded';
+
+            var scriptElem = document.createElement('script');
+            scriptElem.async = true;
+            scriptElem.type = 'text/javascript';
+            scriptElem.src = fullUrl;
+
+            document.body.appendChild(scriptElem);
```

Comment:
And of course the whole point of loading the map script async is to make the tile mode start faster. Imagine if you have like 5-10 apps installed and all of them load tons of extra libraries that they only use during fullscreen mode, thats a lot of unnecessary stuff loaded. Though because you hide the fullscreen-mode stuff with ng-if the map is created instantly so the benefit is kinda lost there.